event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DataEntryUpdateHandler.java

```diff
@@ -22,12 +46,101 @@ public DataEntryUpdateHandler(NotificationClient<NviPublishMessageResponse> snsC
         this.snsClient = snsClient;
     }
 
-    //TODO: Handle all dao types
     @Override
     public Void handleRequest(SQSEvent input, Context context) {
         input.getRecords().stream()
-            .map(SQSEvent.SQSMessage::getBody)
-            .forEach(message -> snsClient.publish(message, CANDIDATE_UPDATE_APPLICABLE));
+            .map(SQSMessage::getBody)
+            .map(this::mapToDynamoDbRecord)
+            .filter(Objects::nonNull)
+            .forEach(this::publishToTopic);
         return null;
     }
+
+    private static boolean isNotCandidateOrApproval(Dao dao) {
+        return !(dao instanceof CandidateDao || dao instanceof ApprovalStatusDao);
+    }
+
+    private static String getTopic(OperationType operationType, Dao dao) {
+        return switch (operationType) {
+            case INSERT, REMOVE -> joinStrings(TOPIC_PREFIX, dao.type(), operationType.toString());
+            case MODIFY -> getUpdateTopic(dao);
+            default -> throw new IllegalArgumentException("Illegal operation type: " + operationType);
+        };
+    }
+
+    private static String getUpdateTopic(Dao dao) {
+        switch (dao.type()) {
+            case CandidateDao.TYPE -> {
+                return getCandidateUpdateTopic(dao);
+            }
+            case ApprovalStatusDao.TYPE -> {
+                return joinStrings(TOPIC_PREFIX, dao.type(), OperationType.MODIFY.toString());
+            }
+            default -> throw new IllegalArgumentException("Illegal dao type: " + dao.type());
+        }
+    }
+
+    private static String getCandidateUpdateTopic(Dao dao) {
+        var candidateDao = (CandidateDao) dao;
+        var applicableTopicString = candidateDao.candidate().applicable() ? APPLICABLE : NOT_APPLICABLE;
+        return joinStrings(TOPIC_PREFIX, dao.type(), OperationType.MODIFY.toString(), applicableTopicString);
+    }
+
+    private static String joinStrings(String... args) {
+        var joiner = new StringJoiner(TOPIC_DELIMITER);
+        for (String arg : args) {
+            joiner.add(arg);
+        }
+        return joiner.toString();
+    }
+
+    private static boolean isUnknownOperationType(OperationType operationType) {
+        return OperationType.UNKNOWN_TO_SDK_VERSION.equals(operationType);
+    }
+
+    private void publishToTopic(DynamodbStreamRecord record) {
```

Comment:
Can you fix occurrences of the `record` keyword where it is used as a variable name?